docs/apache-airflow-providers-apache-spark/connections/spark.rst

```diff
@@ -27,7 +27,7 @@ The Apache Spark connection type enables connection to Apache Spark.
 Default Connection IDs
 ----------------------
 
-Spark Submit and Spark JDBC hooks and operators use ``spark_default`` by default, Spark SQL hooks and operators point to ``spark_sql_default`` by default, but don't use it.
+Spark Submit and Spark JDBC hooks and operators use ``spark_default`` by default, Spark SQL hooks and operators point to ``spark_sql_default`` by default, but its usage is optional.
```

Comment:
change to `use spark_sql_default` instead of `point to spark_sql_default`.
I would completely remove the `, but its usage is optional`, I think it is the normal behaviour for any connection.
```suggestion
Spark Submit and Spark JDBC hooks and operators use ``spark_default`` by default, Spark SQL hooks and operators use ``spark_sql_default`` by default.
```

Reply:
> I would completely remove the `, but its usage is optional`, I think it is the normal behaviour for any connection.

Is it? IIRC the Connection is *not* optional for most hooks; they fail with `AirflowNotFoundException` if a matching Connection is not found.

Reply:
On both Spark connectors, you have the option to send all the parameters and not using the connection as far as I know. Correct me if I am wrong.

Reply:
Hmm yeah `SparkSubmitHook` seems to also follow this approach. I guess that’s an expected approach then!